src/sqlfluff/rules/L008.py

```diff
@@ -34,45 +36,76 @@ class Rule_L008(BaseRule):
         WHERE a IN ('plop',•'zoo')
     """
 
-    def _eval(self, context: RuleContext) -> Optional[LintResult]:
-        """Commas should be followed by a single whitespace unless followed by a comment.
+    def _get_subsequent_whitespace(
+        self,
+        context,
+    ) -> Tuple[Optional[BaseSegment], Optional[BaseSegment]]:
+        """Search forwards through the raw segments for subsequent whitespace.
 
-        This is a slightly odd one, because we'll almost always evaluate from a point a few places
-        after the problem site. NB: We need at least two segments behind us for this to work.
+        Return a tuple of both the trailing whitespace segment and the
+        first non-whitespace segment discovered.
         """
-        if len(context.raw_stack) < 1:
-            return None
+        subsequent_whitespace = None
+        # Get all raw segments and find position of the current comma within the list.
+        file_segment = context.parent_stack[0]
+        raw_segments = file_segment.get_raw_segments()
+        # Raw stack is appropriate as the only segments we can care about are
+        # comma, whitespace, newline, and comment, which are all raw.
+        # Using the raw_segments allows us to account for possible unexpected
+        # parse tree structures resulting from other rule fixes.
+        next_raw_index = raw_segments.index(context.segment) + 1
+        # Iterate forwards over raw segments to find both the whitespace segment and
+        # the first non-whitespace segment.
+        for s in raw_segments[next_raw_index:]:
+            if s.is_meta:
+                continue
+            elif s.is_type("whitespace"):
+                # Capture the whitespace segment.
+                subsequent_whitespace = s
+            else:
+                # We've found a non-whitespace (and non-meta) segment.
+                # Therefore return the stored whitespace segment
+                # and this segment for analysis.
+                return subsequent_whitespace, s
+
+        # If we find ourselves here it's all
+        # whitespace (or nothing) to the end of the file.
+        # Shouldn't be possible to have a comma at the end of a file
```

Comment:
I think it's possible in BigQuery, which allows trailing commas, e.g.:
```
SELECT 
    1,
    2,
```

Reply:
Ah makes sense, I always wondered why L038 was a rule rather than just not parsing. Let me update the comment here and add a unit test for that bigquery example 😄